PreLoad.cmake

```diff
@@ -15,7 +15,7 @@
 if (NOT DEFINED ENV{CLION_IDE} AND NOT DEFINED ENV{XCODE_IDE})
     find_program(NINJA_PATH ninja)
     if (NINJA_PATH)
-        set(CMAKE_GENERATOR "Ninja" CACHE INTERNAL "" FORCE)
+        set(CMAKE_GENERATOR "Ninja" CACHE INTERNAL "")
```

Comment:
Hm, with or without `FORCE` did not make a difference on my system ... I could only build with Ninja.

```
cmake -G"Unix Makefiles" -DENABLE_LIBRARIES=0 -S . -B build
CMake Error: Error: generator : Unix Makefiles
Does not match the generator used previously: Ninja
Either remove the CMakeCache.txt file and CMakeFiles directory or choose a different binary directory.
```
Removing CMakeCache.txt or choosing another build directory didn't help. What helped was commenting out l. 15-20 completely. Could not really find out what was going wrong.

Anyways, I am fine with removing `FORCE` from l. 18.

Reply:
The problem here is in `PreLoad.cmake`, `CMAKE_GENERATOR` hadn't been installed yet, see [Visibility in PreLoad.cmake CMAKE_GENERATOR](https://gitlab.kitware.com/cmake/cmake/-/issues/18449), so I'm afraid there is no sane way to avoid this.

But there is hack - `cmake -G"Unix Makefiles" -DNINJA_PATH=NO ..`

Reply:
Thanks. Guess I shouldn't expect too much from PreLoad.cmake which is completely undocumented 😄  And I was really just trying out if makefile generation works at all, in the end ninja is my build tool of choice.